src/pom.xml

```diff
@@ -1753,7 +1753,31 @@
        </execution>
      </executions>
    </plugin>
-   
+
+   <plugin>
+    <groupId>au.com.acegi</groupId>
+    <artifactId>xml-format-maven-plugin</artifactId>
+    <version>3.1.2</version>
+    <configuration>
+      <includes>**/pom.xml</includes>
```

Comment:
Hum... would this work in a parallel build?

Reply:
I checked the code, theadsafe plugins should be annotated with something like`@Mojo( name = "WhatEver", threadSafe = true)` but that is not the case and I don't see any "syncronized" anywhere'. I can't remember if I tried running with eg. `-T2.2C`, I likely did though out of habit; I'll try and give that a shot.

I don't think there would be any problems seeing that this creates a temp copy that is valid xml (because it uses sax/dom4j) and swaps out the temp file with the formatting target. so in any case the OS/file system should take care of the file integrity I think.
see here: https://github.com/acegi/xml-format-maven-plugin/blob/6a50b3243bff5ec602fea4c7fd6021a93e573500/src/main/java/au/com/acegi/xmlformat/FormatUtil.java#L102-L119 for the writing

Reply:
I wish... but no, see this ticket: https://github.com/revelc/formatter-maven-plugin/issues/433
I'll have a look at the sortpom plugin next weekend (unless someone else is interested and does it before me, if so, please warn others).

Reply:
OK, FWIW; I cherry-picked the first commit of this PR into a new branch (need a bit of work) and ran:

```
git checkout -b xml-formatter-test
git cherry-pick 59a238559bc77e18dabf172a3ef3b400cba37ef1

mvn -B -U -T3.2C -fae -Prelease -f src/pom.xml clean install
mvn -nsu -B -U -T4.2C -fae -DskipTests -Prelease -PcommunityRelease -f src/community/pom.xml clean install
```

There is this warning in the log:


```
[INFO] Building Main Module 2.19-SNAPSHOT                              [19/108]
[INFO] --------------------------------[ jar ]---------------------------------
[WARNING] *****************************************************************
[WARNING] * Your build is requesting parallel execution, but project      *
[WARNING] * contains the following plugin(s) that have goals not marked   *
[WARNING] * as @threadSafe to support parallel building.                  *
[WARNING] * While this /may/ work fine, please look for plugin updates    *
[WARNING] * and/or request plugins be made thread-safe.                   *
[WARNING] * If reporting an issue, report it against the plugin in        *
[WARNING] * question, not against maven-core                              *
[WARNING] *****************************************************************
[WARNING] The following plugins are not marked @threadSafe in Main Module:
[WARNING] au.com.acegi:xml-format-maven-plugin:3.1.2
[WARNING] Enable debug to see more precisely which goals are not marked @threadSafe.
[WARNING] *****************************************************************
```

but no problems; to check I ran the commands as in the `linux_jdk11.yml` workflow on those formatted files


```
mvn -B -U -T3 -fae -Dfmt.skip=true -Prelease -f src/pom.xml clean install
mvn -nsu -B -U -T4 -fae -Dfmt.skip=true -DskipTests -Prelease -PcommunityRelease -f src/community/pom.xml clean install
mvn -f src/pom.xml assembly:single -nsu -N
mvn -f src/community/pom.xml assembly:single -nsu -N

```

And these all report build succes.

Reply:
revelc formatter seems to be using buffered readers and writers to the same file https://github.com/revelc/formatter-maven-plugin/blob/main/src/main/java/net/revelc/code/formatter/FormatterMojo.java at first glance; it's easy to imagine that blow up.


sortpom is marked as `threadSafe` https://github.com/Ekryd/sortpom/blob/8ef972b168390aa9d3993b8d59347e66bf3d49eb/maven-plugin/src/main/java/sortpom/SortMojo.java#L10-L17